python_modules/dagster/dagster_tests/core_tests/resource_tests/pythonic_resources/test_direct_invocation.py

```diff
@@ -411,3 +411,42 @@ def an_asset(context, my_resource: NumResource, my_other_resource: NumResource)
     )
     assert executed["yes"]
     executed.clear()
+
+
+def test_direct_invocation_output_metadata():
+    @asset
+    def my_asset(context):
+        context.add_output_metadata({"foo": "bar"})
+
+    @asset
+    def my_other_asset(context):
+        context.add_output_metadata({"baz": "qux"})
+
+    ctx = build_op_context()
+
+    my_asset(ctx)
+    assert ctx.get_output_metadata("result") == {"foo": "bar"}
+
+    with pytest.raises(
+        DagsterInvariantViolationError,
+        match="attempted to log metadata for output 'result' more than once",
+    ):
+        my_other_asset(ctx)
+
+
+def test_async_assets_with_shared_context():
```

Comment:
@alangenfeld I don't remember the exact specifics of the async test we talked about on friday. do you remember the specific thing we wanted to assert one?

Reply:
the test is to run them in the same event loop, something like [1]